src/transformState.js

```diff
@@ -5,7 +5,41 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const ADD_PROPERTY_TYPE = 'addProperties';
+  const REMOVE_PROPERTY_TYPE = 'removeProperties';
+  const CLEAR_TYPE = 'clear';
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ADD_PROPERTY_TYPE:
+        Object.assign(state, action.extraData);
+
+        break;
+
+      case REMOVE_PROPERTY_TYPE:
+        for (let j = 0; j < action.keysToRemove.length; j++) {
+          const key = action.keysToRemove[j];
+
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
+
+        break;
+
+      case CLEAR_TYPE:
+        for (const key in state) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
```

Comment:
no need to check the the `key` is in `state` object
```suggestion
          delete state[key];
```